microcluster/app.go

```diff
@@ -96,21 +97,25 @@ func (m *MicroCluster) Start(apiEndpoints []rest.Endpoint, schemaExtensions map[
 		return fmt.Errorf("Unable to start daemon: %w", err)
 	}
 
+	stop := sync.OnceFunc(func() {
+		logger.Info("Initiate daemon stop sequence")
+		d.ShutdownDoneCh <- d.Stop()
+	})
+
 	for {
 		select {
+		case err = <-d.ShutdownDoneCh:
+			close(d.ShutdownDoneCh)
+			return err
+		case <-d.ShutdownCtx.Done():
+			stop()
 		case sig := <-sigCh:
 			logCtx := logger.AddContext(logger.Ctx{"signal": sig})
 			logCtx.Info("Received signal")
 			if d.ShutdownCtx.Err() != nil {
 				logCtx.Warn("Ignoring signal, shutdown already in progress")
```

Comment:
we should probably have a `continue` statement here so the loops restarts again without calling `stop()` unnecessarily if a shutdown is already started.